idaes/generic_models/properties/core/eos/virial.py

```diff
@@ -0,0 +1,797 @@
+        for comp in b.parent_block().component_list:
+            cobj = b.parent_block().get_component(comp)
+            if not hasattr(cobj, 'volume_crit'):
+                setattr(cobj, "volume_crit",
+                        Var(doc="Pure component critical Volume",
+                            units=units["volume"] / pyunits.mol))
+                set_param_from_config(cobj, param="volume_crit")
+
+        # kappa binary interaction parameters for i-j molecular pair
+        # for  i=j(like molecules), kappa = 0
+        # for i not equal to  j, kappa is a small positive number.
+        # set to zero if data is unavailable
+
+        try:
+            kappa_data = param_block.config.parameter_data["kappa"]
+            param_block.add_component(
+                'kappa',
+                Var(param_block.component_list,
+                    param_block.component_list,
+                    within=NonNegativeReals,
+                    initialize=kappa_data,
+                    doc='Binary interaction parameters',
+                    units=pyunits.dimensionless))
+        except KeyError:
+            param_block.add_component(
+                'kappa',
+                Var(param_block.component_list,
+                    param_block.component_list,
+                    within=NonNegativeReals,
+                    initialize=0.0,
+                    doc='Binary interaction parameters',
+                    units=pyunits.dimensionless))
+
+        # Since standard property changes of formation are readily available for
+        # Enthalpy(H_ref) and Gibbs free energy (G_ref),
+        # This builds G_ref and set its value from config block, and  then
+        # computes S_ref = (H_ref-G_ref)/T_ref
+        for comp in b.parent_block().component_list:
+            cobj = b.parent_block().get_component(comp)
+            setattr(cobj, "gibbs_mol_form_vap_comp_ref",
+                    Var(doc="Vapor phase molar Gibbs free energy"
+                        "formation @ Tref",
+                        units=units["energy_mole"]))
+
+            if b.parent_block().config.include_enthalpy_of_formation:
+                set_param_from_config(
+                    cobj, param="gibbs_mol_form_vap_comp_ref")
+                S_ref = pyunits.convert(((cobj.enth_mol_form_vap_comp_ref -
+                                          cobj.gibbs_mol_form_vap_comp_ref) /
+                                         b.parent_block().temperature_ref),
+                                        units['entropy_mole'])
+                cobj.entr_mol_form_vap_comp_ref.set_value(value(S_ref))
+            else:
+                cobj.entr_mol_form_vap_comp_ref.set_value(0.0)
+                cobj.gibbs_mol_form_vap_comp_ref.set_value(0.0)
+
+    # -------------------------------------------------------------------------
+    # METHODS FOR PURE SPECIES
+
+    @staticmethod
+    def compress_fact_vap_comp_pure(b, comp):
```

Comment:
How are these methods intended to be used? They are not part of the build-on-demand system, so they will not be called that way, so this suggests that other methods below will be calling them directly. Some explanation of this would be useful here.

Further, why not just build these as part of the `common` method above?

Reply:
lets discuss about these